.github/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/dagger/dagger/.github/internal/dagger"
+)
+
+const (
+	daggerVersion      = "v0.12.5"
+	upstreamRepository = "dagger/dagger"
+	defaultRunner      = "ubuntu-latest"
+)
+
+type CI struct {
+	// +private
+	Gha *dagger.Gha
+}
+
+func New() *CI {
+	ci := new(CI)
+	ci.Gha = dag.Gha(dagger.GhaOpts{
+		DaggerVersion: daggerVersion,
+		PublicToken:   "dag_dagger_sBIv6DsjNerWvTqt2bSFeigBUqWxp9bhh3ONSSgeFnw",
+		Runner:        ci.BronzeRunner(false),
+	})
+	return ci.
+		WithPipeline("Docs", "docs lint", "", false).
+		WithSdkPipelines("python").
+		WithSdkPipelines("typescript").
+		WithSdkPipelines("go").
+		WithSdkPipelines("java").
+		WithSdkPipelines("elixir").
+		WithSdkPipelines("rust").
+		WithSdkPipelines("php")
+}
+
+func (ci *CI) WithPipeline(
+	name string,
+	command string,
+	// +optional
+	runner string,
+	// +optional
+	devEngine bool,
+) *CI {
+	opts := dagger.GhaWithPipelineOpts{
+		OnPushBranches:              []string{"main"},
+		OnPullRequestOpened:         true,
+		OnPullRequestReopened:       true,
+		OnPullRequestSynchronize:    true,
+		OnPullRequestReadyForReview: true,
+		PullRequestConcurrency:      "preempt",
+	}
+	if runner != "" {
+		opts.Runner = runner
+	}
+	if devEngine {
+		opts.DaggerVersion = "."
+	} else {
+		opts.DaggerVersion = daggerVersion
+	}
+	command = fmt.Sprintf("--source=.:default --docker-cfg=file:$HOME/.docker/config.json %s", command)
+	ci.Gha = ci.Gha.WithPipeline(name, command, opts)
+	return ci
+}
+
+func (ci *CI) WithSdkPipelines(sdk string) *CI {
+	ci = ci.
+		WithSdkPipeline(sdk, "lint", false).
+		WithSdkPipeline(sdk, "lint", true).
+		WithSdkPipeline(sdk, "test", false).
+		WithSdkPipeline(sdk, "test", true)
+	// java doesn't have an automated publish step to test
+	if sdk != "java" {
+		ci = ci.
+			WithSdkPipeline(sdk, "test", false).
+			WithSdkPipeline(sdk, "test", true)
+	}
+	return ci
+}
+
+func (ci *CI) WithSdkPipeline(sdk, action string, devEngine bool) *CI {
+	var (
+		name    string
+		command []string
+		runner  string
+	)
+	if devEngine {
+		runner = ci.SilverRunner(true)
+		name = sdk + " sdk: " + action + " (dev-engine)"
+	} else {
+		name = sdk + " sdk: " + action
+	}
+	switch action {
+	case "test", "lint":
+		command = []string{"sdk", sdk, action}
+	case "test-publish":
+		// FIXME: move this logic into a dagger function
+		command = []string{"sdk", sdk, "publish --dry-run=true --tag=$GITHUB_REF"}
+	}
+	return ci.WithPipeline(name, strings.Join(command, " "), runner, devEngine)
+}
+
+// Assemble a runner name for a pipeline
+func (ci *CI) Runner(
+	generation int,
+	daggerVersion string,
+	cpus int,
+	singleTenant bool,
+	noSpot bool,
```

Comment:
It's because "spot instance" became a generic term for a well-known pattern beyond AWS (instances that can be preempted depending on demand and the price you set). Whereas "on demand" is AWS-specific jargon that doesn't carry the same obvious meaning industry-wide.

So `noSpot` will be more obvious to more people.